ocaml/src/osd_access.ml

```diff
@@ -320,17 +320,43 @@ class osd_access
              | Available ->
                 Lwt.return ()
              | ThisAlba osd_id ->
-
-                get_osd_info ~osd_id >>= fun (osd_info, osd_state) ->
-                Osd_state.add_ips_port osd_state ips port;
-                (match extras with
-                 | None -> ()
-                 | Some { used; total; _; } ->
-                    Osd_state.add_disk_usage osd_state (used, total));
-                Osd_state.add_json osd_state json;
-                Osd_state.add_seen osd_state;
-
-                Lwt.return ()
+                begin
+                  get_osd_info ~osd_id >>= fun (osd_info, osd_state) ->
+                  Osd_state.add_ips_port osd_state ips port;
+                  Osd_state.add_json osd_state json;
+                  Osd_state.add_seen osd_state;
+                  let kind',used',total' =
+                    match extras with
+                    | None ->
+                       Osd.Kinetic(ips,port, id), osd_info.Osd.used, osd_info.Osd.total
+                    | Some { used; total; _ } ->
+                       Osd.Asd(ips,port, id), used, total
+
+                  in
+                  let osd_info' =
+                    Osd.{ osd_info with
+                          kind = kind';
+                          total = total'; used = used'
+                    }
+                  in
+                  let () = Osd_state.add_disk_usage osd_state (used', total') in
```

Comment:
used, total.

There's a loop to actively refresh the used,total info by polling the osds (because we want to be multicast independent) at https://github.com/openvstorage/alba/blob/master/ocaml/src/osd_access.ml#L54 .
In case of kinetic this line right here would undo the info that was previously polled.
Actually, come to think of it: in that loop (L54) we should update osd_info.total & used, not here...